Optional `or` branch doesn't define `err`

**V version:** 0.1.8
**OS:** Judge0 playground.

**What did you do?**
```go
fn e() ?int {return error('hi')}

fn main(){
    v := e() or {
	panic(err) // err undefined
	return
    }
    println(v)
}
```

**What did you expect to see?**
`panic('hi')` called. The [docs for optional](https://vlang.io/docs#option) show:
```
resp := http.get(url) or {
	panic(err)
}
```
This doesn't compile. Presumably `err` should define `err := temp_optional.error` at the start of the `or` block. (`Option.error` is not `pub`).
 
**What did you see instead?**
```
panic: main.v:5
undefined: `err`
```

--- a/compiler/parser.v
+++ b/compiler/parser.v
@@ -1128,6 +1128,13 @@ fn (p mut Parser) var_decl() {
 		p.next()
 		p.check(LCBR)
 		p.genln('if (!$tmp .ok) {')
+		p.register_var(Var {
+			name: 'err'
+			typ: 'string'
+			is_mut: false
+			is_used: true
+		})
+		p.genln('string err = $tmp . error;')
 		p.statements()
 		p.genln('$typ $name = *($typ*) $tmp . data;')
 		if !p.returns && p.prev_tok2 != CONTINUE && p.prev_tok2 != BREAK {

--- a/compiler/tests/option_test.v
+++ b/compiler/tests/option_test.v
@@ -1,0 +1,10 @@
+fn opt_err() ?string {return error('hi')}
+
+fn test_err(){
+	v := opt_err() or {
+		assert err == 'hi'
+		return
+	}
+	assert false
+	println(v) // suppress not used error
+}

